source/common/config/delta_subscription_state.cc

```diff
@@ -1,18 +1,36 @@
 #include "common/config/delta_subscription_state.h"
 
+#include "envoy/event/dispatcher.h"
 #include "envoy/service/discovery/v3/discovery.pb.h"
 
 #include "common/common/assert.h"
 #include "common/common/hash.h"
 #include "common/config/utility.h"
+#include "common/runtime/runtime_features.h"
 
 namespace Envoy {
 namespace Config {
 
 DeltaSubscriptionState::DeltaSubscriptionState(std::string type_url,
                                                UntypedConfigUpdateCallbacks& watch_map,
-                                               const LocalInfo::LocalInfo& local_info)
-    : type_url_(std::move(type_url)), watch_map_(watch_map), local_info_(local_info) {}
+                                               const LocalInfo::LocalInfo& local_info,
+                                               Event::Dispatcher& dispatcher)
+    // TODO(snowp): Hard coding VHDS here is temporary until we can move it away from relying on
+    // empty resources as updates.
+    : supports_heartbeats_(!absl::StrContains(type_url, "VirtualHost")),
```

Comment:
Maybe specify the full type_url string instead of doing a substring match here, just in case we ever add a new resource type that happens to have "VirtualHost" as a substring?

Reply:
Yeah that's a good call, will do